### PR TITLE
tutorial works with new modal system

### PR DESCRIPTION
**Pull request by Nick42791**
Tutorial _should_ be fully functional.

note: this pr contains most of Lorenzo's changes up to this point.

Additional stuff:
- Popup modals work differently. this change fixes the issue with the tutorial sliding around.
- Popup modals have a popup animation. this may look weird during the tutorial, so heads up for that.
- Modals now should fit on any screen size, without getting cutoff.
- Fixed way state transitions were happening in tutorial that would cause currentElement null exceptions.

### CODE REVIEW

File: client/src/store/mutationFolder/ui.ts [hunk 1]
```diff
@@ -1,33 +1,67 @@
...
-  state.ui.modalViews.cardModal = payload;
+// NOTE :: CONTROL PROFILE MENU VISIBILITY
+
+function SET_PROFILE_MENU_VISIBILITY(state: State, payload: boolean) {
+  state.userInterface.profileMenuView.visible = payload;
+}
+
+// NOTE :: CONTROL POPUP VISIBILITY
+
+function SET_GROUP_POPUP_VISIBILITY(state: State, payload: boolean) {
```
Comment (chrstngyn) [inline client/src/store/mutationFolder/ui.ts:33]: i like that informational components (popups, menus, modals) may be better to consolidate controls (like visibility) for into one general set visibility function. i would suggest considering ways to consolidate similar behaviors of these informational components. 

however, for the sake of time, this task might be better suited for a refactor after the pretest

File: client/src/stylesheets/root/ContainerDefeat.scss [hunk 2]
```diff
@@ -1,92 +1,166 @@
-.container-defeat{
-    color:white;
-    
-    width: 50rem;
-    margin-left: auto;
-    margin-right: auto;
-    background-color: $space-gray;
-
-    .gameOver-text{
+.c-gameview-defeat {
```
Comment (chrstngyn) [inline client/src/stylesheets/root/ContainerDefeat.scss:1]:
```suggestion
.c-defeat {
```

File: client/src/api/tutorial/steps/trade.ts [hunk 3]
```diff
@@ -185,28 +188,24 @@ const steps: Array<Step> = [
...
     content: "Quiz Question",
     params: {
       placement: RIGHT,
-      tutorialElementId: "trade"
+      tutorialElementId: 'trade',
     },
     stateTransform: [
       {
-        required: false
+        SET_MODAL_HIDDEN:'cool',
```
[the remark this answers is not in the document]
Comment (Nick42791) [reply client/src/api/tutorial/steps/trade.ts:205]: because the data doesn't matter, but it needs a value.

**Comment by Nick42791** [comment]
Hey, @chrstngyn thanks for the review. So, I think pretty much all of the changes requested come directly from @lfaivre's branch. I can go through and fix them, but I'm not sure what he's doing as of now and making these changes will create more conflicts when he pushes (although, since this pr contains all of his changes, I'm not sure if he's pushing at all).

File: client/src/api/tutorial/steps/trade.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (chrstngyn) [reply client/src/api/tutorial/steps/trade.ts:205]: i would just set it to something more intuitive like 'hidden' and not just a random string value

**Comment by chrstngyn** [comment]
since this is already a pretty big PR, i don't think it's a good idea to push additional changes until the changes requested in this PR are resolved and merged. we want to make sure everything is working for the pretest today

**Pull request closed without merging**